var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -81,14 +79,14 @@ class Mfem(Package):
 
     def check_variants(self, spec):
         print spec
-        if '+mpi' in spec and ('+hypre' not in spec or '+metis' not in spec):
+        if '+mpi' in spec and ('hypre' not in spec or 'metis' not in spec):
```

Comment:
I switched it because it was not working for me :)

Seems to me it s the correct logic though, because I want `+mpi` to imply a `hypre` dependence, so I don't have to say `+mpi+hypre`. (In fact `+hypre` alone does not really make a sense as a variant.)